x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/EsqlTestUtils.java

```diff
@@ -1154,6 +1156,31 @@ public static TDigestHolder randomTDigest() {
         return returnValue;
     }
 
+    public static BytesRef randomHistogram() {
+        List<Double> values = ESTestCase.randomList(randomIntBetween(1, 1000), ESTestCase::randomDouble);
+        values.sort(Double::compareTo);
+        // Note - we need the three parameter version of random list here to ensure it's always the same length as values
+        List<Long> counts = ESTestCase.randomList(values.size(), values.size(), ESTestCase::randomNonNegativeLong);
+        BytesStreamOutput streamOutput = new BytesStreamOutput();
+        try {
+            for (int i = 0; i < values.size(); i++) {
+                long count = counts.get(i);
+                assert count >= 0;
+                // we do not add elements with count == 0
```

Comment:
That's just "compression"?

Reply:
Honestly, this is bad copy pasta.  There's no reason we should ever generate a zero in the random test data here, and I'll just update it to not do that.  I believe the original code I copied this from, in the field mapper, does skip empty buckets as a space optimization, but I don't 100% remember the reasoning there.